doc/adrs/195_add_impact_analysis_for_contextual_menu_actions_of_trees.adoc

```diff
@@ -0,0 +1,97 @@
+= ADR-195 - Add impact analysis for contextual menu actions of trees
+
+== Context
+
+A tree contextual menu action can trigger an impact analysis to let the user decide whether they want to execute the tool.
+
+== Decision
+
+We will add an API to run impact analysis on trees and factorize common code between diagram and tree impact analysis.
+
+
+=== Frontend
+
+We will move the following types from `sirius-components-diagrams` to `sirius-components-core` to make the impact analysis dialog reusable in non-diagram contexts:
```

Comment:
In this ADR I tried to see which parts of the impact analysis could be factorized and reused in both diagrams and trees. I don't know if this fits with the vision you have for this feature. If it is not the case, I can rework the ADR to basically duplicate everything and make it tree-specific.

Reply:
This can be moved but I think that some concepts will have to be changed because some of those files are tightly connected to the concept of tool for example.

Reply:
We could maybe extract `useInvokeImpactAnalysis` from its context, and call it before showing the impact analysis dialog. This way we'll use the same context to display the dialog, but the way the information is retrieved can be defined at the representation-level (with toolId for diagrams and menuEntryId for trees). 

I have added a generic sentence in the ADR to trace that more refactoring may be required, but I am not sure I have the keys to know exactly its scope at the moment. If the solution above looks good to you I can put it in the ADR.